app/services/samples/metadata/update_service.rb

```diff
@@ -69,14 +67,10 @@ def validate_metadata_param
               I18n.t('services.samples.metadata.empty_metadata', sample_name: @sample.name)
       end
 
-      def transform_metadata_keys
-        # Without transforming and downcasing keys,
-        # issues with overwritting can occur and multiples of the same key can appear
-        @metadata = @metadata.transform_keys { |key| key.to_s.downcase }
-      end
-
       def perform_metadata_update
         @metadata.each do |key, value|
+          key = key.to_s.downcase
+          value = value.to_s # remove data types
```

Comment:
Can you strip white space from the keys and values? I noticed that white space is also getting included when trying to look up the sample id from an xlsx file.